book/src/getting-started/1-qobjects-in-rust.md

```diff
@@ -27,32 +27,31 @@ Therefore it is an ideal candidate to replace C++ for writing the back-end busin
 
 However, C++ as well as QML still have their place in Qt applications.
 For that reason Rust, C++, and QML should all be able to be used to complement each other.
-CXX-Qt aims to make it easy to integrate all three languages easily with each other, through the use of Qt's [meta object system](https://doc.qt.io/qt-5/metaobjects.html).
-
-Qt's design is inherently object-oriented, which is true both for C++ and QML as well.
-Therefore, in order to integrate well with Qt, Rust needs to be able to extend the Qt object system with its own QObject subclasses and instances.
-This is exactly what CXX-Qt allows you to do.
-
-As Rust doesn't offer classes with inheritance and polymorphism, CXX-Qt uses macros when defining new QObject subclasses.
-
-These CXX-Qt modules consist of multiple parts:
-- A struct marked with a `#[cxx_qt::qobject]` macro
-    - Defines the C++ QObject subclass name
-    - A normal Rust struct.
-    - One struct instance is created per class instance.
-    - Defines which Properties will be in the QObject subclass.
-    - Can contains any Rust-only data.
-    - Needs to implement the `Default` trait.
-- The `impl` of the `#[cxx_qt::qobject]` marked struct (optional):
-    - Contains any Rust code.
-    - Functions marked with `#[qinvokable]` will be callable from QML and C++.
-- The `Signal` enum
-    - A normal Rust enum.
-    - Defines signals that are added to the QObject class
+CXX-Qt aims to make it easy to integrate all three languages with each other, through the use of Qt's [meta object system](https://doc.qt.io/qt-5/metaobjects.html) and [CXX](https://cxx.rs).
+
+As CXX-Qt is largely an expansion on top of CXX, you should make yourself familiar with CXX first.
+You can read their excellent documentation here:
+> [https://cxx.rs/](https://cxx.rs)
+
+CXX already provides idiomatic interaction between C++ and Rust, from Rust, as well as from C++.
+CXX-Qt builds on this foundation, to allow you to define Qt-specific concepts in Rust.
+
+These concepts include:
+- Custom QObject classes
+- Properties
+- Signals
+- Invokables/Slots
+
+As with CXX, to use these features you mark a Rust module with an attribute macro (`#[cxx_qt::bridge]`).
+Then you can use the afformentioned features with the help of more macros.
+- `#[cxx_qt::qobject]` - Expose a Rust struct to Qt as a QObject subclass.
+    - `#[qproperty]` - Expose a field of the Rust struct to QML/C++ as a [`Q_PROPERTY`](https://doc.qt.io/qt-6/qtqml-cppintegration-exposecppattributes.html#exposing-properties).
+    - `#[qinvokable]` - Expose a function on the QObject to QML and C++ as a [`Q_INVOKABLE`](https://doc.qt.io/qt-6/qtqml-cppintegration-exposecppattributes.html#exposing-methods-including-qt-slots).
+- `#[cxx_qt::signals(T)]` - Use an enum to define the Signals of a QObject T.
```

Comment:
these items being listed like this makes me wonder.... should it be `cxx_qt::qsignals` ? to match `Q_SIGNALS` ? (easy to change if we want to do this quickly in code...)

Reply:
And https://doc.qt.io/qt-6/qtqml-cppintegration-exposecppattributes.html#exposing-signals should also be linked in this line like the others with the keyword `Q_SIGNALS` ?

Reply:
Hm, that's an interesting point.
Usually I just use `signals:` in my C++ code, not `Q_SIGNALS`.

But for consistency with the other cxx_qt macros I think changing to `cxx_qt::qsignals(T)` is better.
I'll add the link and update the getting started guide.

Reply:
Updated